src/hotspot/share/oops/method.cpp

```diff
@@ -1011,9 +1011,9 @@ void Method::set_native_function(address function, bool post_event_flag) {
   // This function can be called more than once. We must make sure that we always
   // use the latest registered method -> check if a stub already has been generated.
   // If so, we have to make it not_entrant.
-  CompiledMethod* nm = code(); // Put it into local variable to guard against concurrent updates
+  CodeBlob* nm = code();       // Put it into local variable to guard against concurrent updates
   if (nm != NULL) {
-    nm->make_not_entrant();
+    nm->as_compiled_method()->make_not_entrant();
```

Comment:
Ah, actually, this code is just trying to prevent the existing native wrapper from being used in the future. Since it embeds the target native function as a constant ([compile_method](https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/x86/sharedRuntime_x86_64.cpp#L1927))

Forget what I said :) (*sigh* Friday evenings)